### PR TITLE
Fix python_modular warnings by adding API documentation

### PR DESCRIPTION
**Pull request by yorkerlin**

### CODE REVIEW

**Comment by karlnapf** [comment]
Thanks wu

Please make sure you always do pull as
'''git pull --rebase upstream develop '''

To avoid the merge message in the patch above, keeps the logs clean

**Comment by yorkerlin** [comment]
@karlnapf 
Any update about the error?

**Comment by karlnapf** [comment]
checking

**Comment by yorkerlin** [comment]
I merge this since the bug is not related to this PR.

**Comment by karlnapf** [comment]
Ah great does trying this again. How many warnings are left?

File: src/shogun/optimization/ElasticNetPenalty.h [hunk 1]
```diff
@@ -92,11 +102,20 @@ class ElasticNetPenalty: public SparsePenalty
 		return grad;
 	}
 
-	virtual void set_rounding_eplison(float64_t eplison)
+	/** Set the rounding epslion for L1 penalty
+	 *
+	 * @param epslion rounding epslion
+	 *
```
Comment (iglesias) [inline src/shogun/optimization/ElasticNetPenalty.h]: I'd say let's try to be consistent with the doc in the other methods and remove this line.

File: src/shogun/optimization/ElasticNetPenalty.h [hunk 2]
```diff
@@ -92,11 +102,20 @@ class ElasticNetPenalty: public SparsePenalty
...
+	 */
+	virtual void set_rounding_epslion(float64_t epslion)
 	{
-		m_l1_penalty->set_rounding_eplison(eplison);
+		m_l1_penalty->set_rounding_epslion(epslion);
 	}
 
+	/** Do proximal projection/operation in place
```
Comment (iglesias) [inline src/shogun/optimization/ElasticNetPenalty.h]: New line to be consistent between this line and the next. There are similar cases to this case and the one above in other methods in this file and files below.

File: src/shogun/optimization/ElasticNetPenalty.h [hunk 3]
```diff
@@ -92,11 +102,20 @@ class ElasticNetPenalty: public SparsePenalty
 		return grad;
 	}
 
-	virtual void set_rounding_eplison(float64_t eplison)
+	/** Set the rounding epslion for L1 penalty
+	 *
+	 * @param epslion rounding epslion
+	 *
+	 */
+	virtual void set_rounding_epslion(float64_t epslion)
```
Comment (iglesias) [inline src/shogun/optimization/ElasticNetPenalty.h]: Maybe typo: epsilon? Same in the rest of this file.

File: src/shogun/optimization/L1Penalty.h [hunk 4]
```diff
@@ -75,20 +79,20 @@ class L1Penalty: public SparsePenalty
 	virtual float64_t get_penalty_gradient(float64_t variable,
 		float64_t gradient_of_variable) {return 0.0;}
 
-	/** Set the rounding eplison
+	/** Set the rounding epslion
 	 *
-	 * @param eplison rounding eplison
+	 * @param epslion rounding epslion
```
Comment (iglesias) [inline src/shogun/optimization/L1Penalty.h]: Maybe you mean epsilon and not epslion? Almost there :-)

File: tests/unit/optimization/StochasticMinimizers_unittest.cc [hunk 5]
```diff
@@ -1032,7 +1032,7 @@ TEST(L1PenaltyForTG, test1)
 	opt->set_penalty_weight(0.01);
 	L1Penalty* penalty_type=new L1PenaltyForTG();
 
-	penalty_type->set_rounding_eplison(0);
+	penalty_type->set_rounding_epslion(0);
```
Comment (iglesias) [inline tests/unit/optimization/StochasticMinimizers_unittest.cc]: maybe epsilon.

Comment (karlnapf) [inline tests/unit/optimization/StochasticMinimizers_unittest.cc]: Thanks for the feedback, @iglesias :)

Comment (iglesias) [inline tests/unit/optimization/StochasticMinimizers_unittest.cc]: No problemo, amigo, my pleasure :dancer:

**Comment by yorkerlin** [comment]
many thanks to @iglesias 
I will correct these typos and the remaining warnings related to the optimization framework.

**Comment by iglesias** [comment]
No problem, @yorkerlin, you are welcome. Thanks to you for making all these cleanups.

**Pull request merged**